msbuild/Xamarin.MacDev.Tasks.Core/Tasks/XcodeCompilerToolTask.cs

```diff
@@ -125,6 +116,22 @@ static ProcessStartInfo GetProcessStartInfo (IDictionary<string, string> environ
 			return startInfo;
 		}
 
+		static bool? translated;
+
+		[DllImport ("/usr/lib/libSystem.dylib", SetLastError = true)]
+		unsafe static extern int sysctlbyname (/* const char */ [MarshalAs (UnmanagedType.LPStr)] string property, void *oldp, void *oldlenp, void *newp, /* size_t */ long newlen);
+
+		// https://developer.apple.com/documentation/apple_silicon/about_the_rosetta_translation_environment
+		static unsafe bool IsTranslated ()
+		{
+			if (translated == null) {
+				long result = 0;
+				long size = sizeof (long);
+				translated = sysctlbyname ("sysctl.proc_translated", (void*)&result, (void*)&size, null, 0) != -1;
```

Comment:
I think this would look a bit better:

```csharp
unsafe static extern int sysctlbyname (/* const char */ [MarshalAs (UnmanagedType.LPStr)] string property, ref long oldp, ref long oldlenp, IntPtr newp, /* size_t */ long newlen);

translated = syctlbyname ("sysctl.proc_translated", ref result, ref size, IntPtr.Zero, 0);
```

and it's not unsafe anymore.